src/aiida/orm/nodes/process/process.py

```diff
@@ -151,6 +164,10 @@ def __str__(self) -> str:
 
         return f'{base}'
 
+    @classproperty
+    def _hash_ignored_attributes(cls) -> Tuple[str, ...]:  # noqa: N805
+        return super()._hash_ignored_attributes + ('metadata_inputs',)
```

Comment:
Late to the party, but I guess instead of the hardcoded string `'metadata_inputs'` it should be `self.METADATA_INPUTS_KEY`? Or would that have some unintended consequences?

Trying to understand if this PR affects the `metadata` attrs on CalcJobNodes, since I noticed with AiiDA 2.4.3 that a cached Node is not used when I submit a calculation with different number of MPI processes.

CC @sphuber

Reply:
Fair enough. There's definitely trade-off there. For my aiidalab-related use, such difference are unlikely to to matter to users, I'll look into how to tweak that behaviour. Thanks!